src/Swagger/Serializer/ApiGatewayNormalizer.php

```diff
@@ -67,18 +70,18 @@ public function normalize($object, $format = null, array $context = [])
                             unset($data['paths'][$path][$operation]['parameters'][$key]);
                         }
                         if (isset($parameter['schema']['$ref']) && !preg_match('/^#\/definitions\/[A-z]+$/', $parameter['schema']['$ref'])) {
-                            $data['paths'][$path][$operation]['parameters'][$key]['schema']['$ref'] = str_replace(['-', '_'], '', $parameter['schema']['$ref']);
+                            $data['paths'][$path][$operation]['parameters'][$key]['schema']['$ref'] = str_replace(['-', ':', '_'], '', $parameter['schema']['$ref']);
```

Comment:
We should not hardcode the characters to replace. Just use `preg_replace` with an appropriate pattern.

Reply:
I'm totally with you, but it wouldn't fit here as Model names can only contain alphanumeric characters. However, this affects the whole $ref string and not just the Model name.

Reply:
What I mean is, we remove all non-alphanumeric characters using `preg_replace`. :smile:

Reply:
I understand what you mean, but removing all non-alpha chars in the $ref string would strip out needed characters (like #, /). I actually would prefer to have a Normalizer for the Model name itself, rather than iterating through all ref strings (with the chance of missing some) (would probably fix DTO name generation with custom input too (the hash after the colon)).

Reply:
Okay, I get what you mean now...

Reply:
But actually, how do we know that these characters aren't used in other parts of the `$ref`? After all, `$ref` can be a URI, right?

Reply:
$ref can be an URI, right. I'm not quite sure what you mean, I think a good solution would be to provide a function that can modify the Model names which the normalizer uses.

Reply:
I just saw #3198 - This could be a step in the right direction, what do you think?

Reply:
If `$ref` could be any URI, and not just the ones we generate, then it's unsafe to perform any kind of search and replace. :smile:

On the other hand, I don't see any relevance between #3198 and this issue.

Reply:
I see the problem, however in it's current version the ApiGatewayNormalizer is non-working, so this commit would at least allow users to use this feature (which is also referenced in the docs).

Reply:
Can we check if the `$ref` value starts with `#/` before doing this replacement? :smile:

Reply:
I could do a check if it starts with #/, then I would strip everything until the last / and pass the model name to a custom function. Sounds good?

Reply:
Never mind, I've just noticed we already have a `preg_match` above. We just need to update the pattern to allow alphanumeric for the definition name, instead of only alpha.

Reply:
Since we're already doing a `preg_match`, we can get the matches from there! So, no separate function.